contrib/re_parser/src/re_parser.nit

```diff
@@ -0,0 +1,209 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module re_parser
+
+import nitcc::autom
+import re_parser_lexer
+import re_parser_parser
+
+class REVisitor
+	super Visitor
+
+	var nfa = new Automaton
+
+	fun start(n: Node) do
+		enter_visit(n)
+	end
+
+	redef fun visit(n) do n.accept_revisitor(self)
+end
+
+redef class Node
+	fun accept_revisitor(v: REVisitor) do visit_children(v)
+
+	# Build the NFA of the regular expression
+	fun make_nfa: Automaton do
+		print inspect
+		abort
+	end
+end
+
+redef class Nre
+	redef fun accept_revisitor(v) do
+		v.nfa = make_nfa
+	end
+
+	redef fun make_nfa do
+		var a = new Automaton
+		for child in children do
+			if child == null then continue
+			a.concat(child.make_nfa)
+		end
+		return a
+	end
+end
+
+redef class Nre_char
+	redef fun make_nfa do
+		return new Automaton.atom(n_char.text.chars.first.code_point)
+	end
+end
+
+redef class Nre_alter
+	redef fun make_nfa
+	do
+		var a = n_re.make_nfa
+		var b = n_re2.make_nfa
+		a.alternate(b)
+		return a
+	end
+end
+
+redef class Nre_conc
+	redef fun make_nfa
+	do
+		var a = n_re.make_nfa
+		var b = n_re2.make_nfa
+		a.concat(b)
+		return a
+	end
+end
+
+redef class Nre_star
+	redef fun make_nfa
+	do
+		var a = n_re.make_nfa
+		a.close
+		return a
+	end
+end
+
+redef class Nre_ques
+	redef fun make_nfa
+	do
+		var a = n_re.make_nfa
+		a.optionnal
+		return a
+	end
+end
+
+redef class Nre_plus
+	redef fun make_nfa
+	do
+		var a = n_re.make_nfa
+		a.plus
+		return a
+	end
+end
+
+redef class Nre_par
+	redef fun make_nfa
+	do
+		return n_re.make_nfa
+	end
+end
+
+redef class Automaton
+	# Produce a graphvis file for the automaton
+	redef fun to_dot do
```

Comment:
I prefer that you improve the original to_dot.
Your LR orientation and ε are fine for the default.

Maybe just add an optional parameter for the edge fusion or split?